cql3/restrictions/statement_restrictions.cc

```diff
@@ -1340,5 +1340,75 @@ const single_column_restrictions::restrictions_map& statement_restrictions::get_
     return single_restrictions->restrictions();
 }
 
+void statement_restrictions::prepare_indexed(const schema& idx_tbl_schema, bool is_local) {
+    if (is_local || !_partition_range_is_simple) {
+        return;
+    }
+    // This assumes that a base-table partition column cannot be the indexed column:
+    _idx_tbl_ck_prefix = std::vector<expr::expression>(1 + _schema->partition_key_size());
+    _idx_tbl_ck_prefix->reserve(idx_tbl_schema.clustering_key_size());
```

Comment:
In the previous line, you initialized it to a vector of size 1 + partition_key_size. In this line, you reserve to a different size? Maybe you want to reserve an additional amount?

Reply:
Yes, that's what I meant, thanks for catching it.  Fixed in v3.